pylint/checkers/variables.py

```
@@ -1826,16 +1831,16 @@ def _store_type_annotation_node(self, type_annotation):
             self._type_annotation_names.append(type_annotation.name)
             return
 
-        if not isinstance(type_annotation, astroid.Subscript):
+        if not hasattr(type_annotation, "nodes_of_class"):
```

[major] This helper lives in the internal logic layer, so it should only ever receive astroid nodes. Switching from the old  guard to a  probe reintroduces duck typing and creates a silent fallback path for unexpected data. Please keep the contract explicit by checking for the concrete astroid base type instead (e.g. ) so we fail loudly if the invariant is broken.